File: components/physics/src/main/java/org/storm/physics/constants/Vectors.java
package org.storm.physics.constants;

import org.storm.physics.math.Vector;

/**
 * A Collections of helpful predefined Vectors
 */
public class Vectors {

  public static final Vector ZERO_VECTOR = new Vector(0, 0);
  public static final Vector UNIT_NORTH = new Vector(0, -1);
  public static final Vector UNIT_NORTH_EAST = new Vector(1, -1);
  public static final Vector UNIT_EAST = new Vector(1, 0);
  public static final Vector UNIT_SOUTH_EAST = new Vector(1, 1);
  public static final Vector UNIT_SOUTH = new Vector(0, 1);
  public static final Vector UNIT_SOUTH_WEST = new Vector(-1, 1);
  public static final Vector UNIT_WEST = new Vector(-1, 0);
  public static final Vector UNIT_NORTH_WEST = new Vector(-1, -1);

}

File: components/physics/src/main/kotlin/org/storm/physics/constants/Vectors.kt
package org.storm.physics.constants

import org.storm.physics.math.Vector

/**
 * A Collections of helpful predefined Vectors
 */
object Vectors {
  @JvmField
  val ZERO_VECTOR = Vector(0.0, 0.0)

  @JvmField
  val UNIT_NORTH = Vector(0.0, -1.0)

  @JvmField
  val UNIT_NORTH_EAST = Vector(1.0, -1.0)

  @JvmField
  val UNIT_EAST = Vector(1.0, 0.0)

  @JvmField
  val UNIT_SOUTH_EAST = Vector(1.0, 1.0)

  @JvmField
  val UNIT_SOUTH = Vector(0.0, 1.0)

  @JvmField
  val UNIT_SOUTH_WEST = Vector(-1.0, 1.0)

  @JvmField
  val UNIT_WEST = Vector(-1.0, 0.0)

  @JvmField
  val UNIT_NORTH_WEST = Vector(-1.0, -1.0)
}
